pkg/action/upgrade.go

```diff
@@ -475,6 +477,7 @@ func (u *Upgrade) failRelease(rel *release.Release, created kube.ResourceList, e
 		rollin.Recreate = u.Recreate
 		rollin.Force = u.Force
 		rollin.Timeout = u.Timeout
+		rollin.RollbackTimeout = u.UpgradeTimeout / 2
```

Comment:
UpgradeTimeout is never assigned a value, so wouldn't this division operation fail?

(I have very little go experience so I might be missing something here)

Reply:
Wait a second - https://stackoverflow.com/a/4223332

> because by default all variables (including dynamically allocated ones) are set to “zero values” according to type (numbers zero, references nil)

weeeiirrrrddddd

So this would be 0/2, which is 0. Would this cause the rollback to immediately timeout?

Reply:
@caleb15 I've added some defaults to deal with the upgrade timeout dangling issue. Please review,